### PR TITLE
Add testing for punet model variations.

### PR DESCRIPTION
**Pull request by saienduri**
So far, we have been testing the fp16 model for unet in our regression testing. This commit adds testing for our latest int8 punet model (fp16 attention and fp8 attention variants) that we have switched over to using for sdxl. It is more performant and efficient than the fp16 model pulled from huggingface. The model is defined over here: https://github.com/nod-ai/SHARK-Platform/tree/main/sharktank/sharktank/models/punet. For now, I have added testing for compilation and accuracy for rocm so that changes are limited to what we care about and our testing daily. Once we get back on the sdxl perf track, I can also add this to the benchmarking suite, but for now we are keeping track here for the different variations: https://github.com/nod-ai/2024q2-sdxl-mlperf-sprint/actions. We also remove all uses of a spec file in this special model testing suite.

### CODE REVIEW

File: build_tools/pkgci/external_test_suite/attention_and_matmul_punet_spec.mlir
Comment (ScottTodd) [inline build_tools/pkgci/external_test_suite/attention_and_matmul_punet_spec.mlir:1]: Are these models still using transform dialect spec files? Can we test without the file, or share the file between different tests?

I don't want us to host another 600 line spec file in this repository for an experimental test if we can avoid it.

File: experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py [hunk 1]
```diff
@@ -104,11 +206,20 @@ def SDXL_UNET_COMMON_RUN_FLAGS(
...
 
+FP16_UNET_FLAGS = [
+    f"--iree-codegen-transform-dialect-library={iree_test_path_extension}/attention_and_matmul_spec.mlir",
+    "--iree-preprocessing-pass-pipeline=builtin.module(iree-preprocessing-transpose-convolution-pipeline,iree-preprocessing-pad-to-intrinsics)",
+]
+
+INT8_PUNET_FLAGS = [
+    f"--iree-codegen-transform-dialect-library={iree_test_path_extension}/attention_and_matmul_punet_spec.mlir",
+    "--iree-preprocessing-pass-pipeline=builtin.module(util.func(iree-global-opt-raise-special-ops, iree-flow-canonicalize), iree-preprocessing-transpose-convolution-pipeline, iree-preprocessing-pad-to-intrinsics, util.func(iree-preprocessing-generalize-linalg-matmul-experimental))",
+]
```
Comment (ScottTodd) [inline experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py]: How load-bearing are these preprocessing pass pipelines? Can we test with closer to default flags instead?

File: experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py [hunk 2]
```diff
@@ -176,7 +287,7 @@ def test_compile_unet_pipeline_rocm(sdxl_unet_pipeline_mlir):
 def test_compile_unet_rocm(sdxl_unet_mlir):
     VmfbManager.sdxl_unet_rocm_vmfb = iree_compile(
         sdxl_unet_mlir,
-        ROCM_COMPILE_FLAGS,
+        ROCM_COMPILE_FLAGS + FP16_UNET_FLAGS,
```
Comment (ScottTodd) [inline experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py]: Now that there are several variations, please put the data types in all test names, and generated artifact names.

File: experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py (hunk 1 above)
Comment (saienduri) [reply experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py]: I tried without the preprocessing pass pipeline at all and with the simpler flag being used for fp16. In both scenarios, it is failing accuracy :(

Comment (ScottTodd) [reply experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py]: We need to drive towards the compiler producing correct programs without a soup of flags and extra configuration files. If the test is failing, it's failing.

File: build_tools/pkgci/external_test_suite/attention_and_matmul_punet_spec.mlir
Comment (saienduri) [reply build_tools/pkgci/external_test_suite/attention_and_matmul_punet_spec.mlir:1]: Did some experiments, and we don't need any transform dialect spec files anymore for either unet or punet 😄. One step in the right direction. Looks like the default C++ pipelines are good enough now

File: experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py (hunk 1 above)
Comment (saienduri) [reply experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py]: Yeah, agreed. At the moment, the preprocess flag is needed for unet or the benchmark for unet is wildly off. For punet, it fails accuracy without its preprocess flag. With how much we care about these models, if we can have these testing running with some less than ideal flags, we can save tons of debugging time and hence spend more time on improving perf. Looks like we are slowly making steps toward the basic config with no longer needing spec files. Hopefully, we can get rid of this soon too

File: build_tools/pkgci/external_test_suite/attention_and_matmul_punet_spec.mlir
Comment (ScottTodd) [reply build_tools/pkgci/external_test_suite/attention_and_matmul_punet_spec.mlir:1]: Nice! Thanks for checking, great cleanup.

File: experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py (hunk 1 above)
Comment (ScottTodd) [reply experimental/regression_suite/shark-test-suite-models/sdxl/test_unet.py]: Okay, fair enough. Thanks for checking. FYI @Groverkss (you recently filed some issues about making some other flags not required)

**Review by ScottTodd — APPROVED** [review]
These new tests may add a few minutes to the MI250 and MI300 jobs. They are still fast enough IMO, but worth keeping an eye on.

**Pull request merged**